tools-v2/pkg/cli/command/curvebs/update/leader_schedule/leader_schedule.go

```diff
@@ -0,0 +1,164 @@
+}
+
+var _ basecmd.FinalCurveCmdFunc = (*LeaderScheduleCommand)(nil) // check interface
+
+const (
+	leaderScheduleExample = `$ curve bs update leader-schedule --logicalpoolid 1
+$ curve bs update leader-schedule --all`
+)
+
+func NewLeaderScheduleCommand() *cobra.Command {
+	return NewUpdateLeaderScheduleCommand().Cmd
+}
+
+func (lCmd *LeaderScheduleCommand) AddFlags() {
+	config.AddBsMdsFlagOption(lCmd.Cmd)
+	config.AddRpcRetryTimesFlag(lCmd.Cmd)
+	config.AddRpcTimeoutFlag(lCmd.Cmd)
+
+	config.AddBsLogicalPoolIdOptionFlag(lCmd.Cmd)
+	config.AddBsAllOptionFlag(lCmd.Cmd)
+}
+
+func NewUpdateLeaderScheduleCommand() *LeaderScheduleCommand {
+	sCmd := &LeaderScheduleCommand{
+		FinalCurveCmd: basecmd.FinalCurveCmd{
+			Use:     "leader-schedule",
+			Short:   "rapidly transfer leader",
+			Example: leaderScheduleExample,
+		},
+	}
+
+	basecmd.NewFinalCurveCli(&sCmd.FinalCurveCmd, sCmd)
+	return sCmd
+}
+
+func (lCmd *LeaderScheduleCommand) Init(cmd *cobra.Command, args []string) error {
+	mdsAddrs, err := config.GetBsMdsAddrSlice(lCmd.Cmd)
+	if err.TypeCode() != cmderror.CODE_SUCCESS {
+		return err.ToError()
+	}
+
+	header := []string{cobrautil.ROW_RESULT, cobrautil.ROW_REASON}
+	lCmd.SetHeader(header)
+
+	timeout := config.GetFlagDuration(lCmd.Cmd, config.RPCTIMEOUT)
+	retryTimes := config.GetFlagInt32(lCmd.Cmd, config.RPCRETRYTIMES)
+
+	// if flag all not changed, it must be false
+	all := config.GetBsFlagBool(lCmd.Cmd, config.CURVEBS_ALL)
+	islogicalPoolIDChanged := config.GetBsFlagChanged(lCmd.Cmd, config.CURVEBS_LOGIC_POOL_ID)
+	if !all && !islogicalPoolIDChanged {
+		return fmt.Errorf("all or logicalpoolid is required")
+	}
+
+	var logicalPoolID uint32
+	if all {
+		logicalPoolID = 0
+	} else {
+		logicalPoolID = config.GetBsFlagUint32(lCmd.Cmd, config.CURVEBS_LOGIC_POOL_ID)
+		if logicalPoolID == 0 {
+			return fmt.Errorf("please set flag all true instead of setting flag logicalpoolid 0")
+		}
+	}
+
+	lCmd.Rpc = &UpdateLeaderScheduleRpc{
+		Request: &schedule.RapidLeaderScheduleRequst{
+			LogicalPoolID: &logicalPoolID,
+		},
+		Info: basecmd.NewRpc(mdsAddrs, timeout, retryTimes, "RapidLeaderSchedule"),
+	}
+
+	return nil
+}
+
+func (lCmd *LeaderScheduleCommand) Print(cmd *cobra.Command, args []string) error {
+	return output.FinalCmdOutput(&lCmd.FinalCurveCmd, lCmd)
+}
+
+func (lCmd *LeaderScheduleCommand) RunCommand(cmd *cobra.Command, args []string) error {
+	result, err := basecmd.GetRpcResponse(lCmd.Rpc.Info, lCmd.Rpc)
+	if err.TypeCode() != cmderror.CODE_SUCCESS {
+		return err.ToError()
+	}
+	out := make(map[string]string)
+	if response, ok := result.(*schedule.RapidLeaderScheduleResponse); ok {
+		if response.GetStatusCode() != 0 {
+			out[cobrautil.ROW_RESULT] = cobrautil.ROW_VALUE_FAILED
+			out[cobrautil.ROW_REASON] = statuscode.ScheduleStatusCode_name[*response.StatusCode]
+		} else {
+			out[cobrautil.ROW_RESULT] = cobrautil.ROW_VALUE_SUCCESS
+			out[cobrautil.ROW_REASON] = cobrautil.ROW_VALUE_NULL
+		}
```

Comment:
This can be handled by adding `ErrBsRapidLeaderSchedule`.

```suggestion
        err = cmderr.ErrBsRapidLeaderSchedule(response.GetStatusCode())
		if err.TypeCode() != cmderr.CODE_SUCCESS {
			out[cobrautil.ROW_RESULT] = cobrautil.ROW_VALUE_FAILED
		} else {
			out[cobrautil.ROW_RESULT] = cobrautil.ROW_VALUE_SUCCESS
		}
        out[cobrautil.ROW_REASON] = err.Message
```